api/integrations/launch_darkly/services.py

```diff
@@ -0,0 +1,297 @@
+from contextlib import contextmanager
+from typing import TYPE_CHECKING
+
+from django.conf import settings
+from django.core import signing
+from django.utils import timezone
+from requests.exceptions import HTTPError, RequestException
+
+from environments.models import Environment
+from features.feature_types import MULTIVARIATE, STANDARD
+from features.models import Feature, FeatureState, FeatureStateValue
+from features.multivariate.models import (
+    MultivariateFeatureOption,
+    MultivariateFeatureStateValue,
+)
+from features.value_types import STRING
+from integrations.launch_darkly import types as ld_types
+from integrations.launch_darkly.client import LaunchDarklyClient
+from integrations.launch_darkly.constants import (
+    LAUNCH_DARKLY_IMPORTED_TAG_COLOR,
+)
+from integrations.launch_darkly.models import (
+    LaunchDarklyImportRequest,
+    LaunchDarklyImportStatus,
+)
+from projects.tags.models import Tag
+
+if TYPE_CHECKING:  # pragma: no cover
+    from projects.models import Project
+    from users.models import FFAdminUser
+
+
+def _sign_ld_value(value: str, user_id: object) -> str:
```

Comment:
user_id will be int, right?

Reply:
Yes, but the function should not care about having an `int` here. Django pks can be anything so I decided against using the `int` annotation.

Reply:
I've changed it, even though this is definitely not the place to guard against non-id PKs.